src/utils/subscription-manager.ts

```diff
@@ -25,6 +25,23 @@ export class SubscriptionManager {
     this.socket.on(event, callback)
   }
 
+  /**
+   * Remove an event subscription
+   * @param {string} event event name
+   */
+  removeSubscription (event: string): void {
+    const ind: number = this.listeners.findIndex(({ key }) => {
```

Comment:
If you have multiple listeners for the same event this will only remove the first, suggest doing a `filter` over a `findIndex` and `forEach` over the whatever is found.